src/rpc/request.cpp

```diff
@@ -142,6 +150,10 @@ bool GetAuthCookie(std::string *cookie_out)
     std::ifstream file;
     std::string cookie;
     fs::path filepath = GetAuthCookieFile();
+    if (filepath.empty()) {
+        LogDebug(BCLog::RPC, "RPC authentication cookie file reading is disabled.");
+        return false;
```

Comment:
why is this false and not an empty optional? As mentioned before, I don't really see the line

Reply:
`GetAuthCookieFile()` internally uses `GetPathArg()` and follows it's pattern (predating this PR):
https://github.com/bitcoin/bitcoin/blob/7590e93bc73b3bbac641f05d490fd5c984156b33/src/common/args.cpp#L272-L274

But might change to `optional` if you insist. Tried out changing it in 01fa9d41d12a68f88f6251ae43234dd250a53bd5. Works fairly well, even uncovered a minor pre-existing bug in that if `fs::permissions` fails, we currently print out the wrong filename. With that bug it feels like it could arguably be its own micro-PR.

Reply:
I'm fine with doing it in a different PR - will leave it up to you